### PR TITLE
[JBTM-4002] Removed nexus staging plugin and added requirements to deploy to Maven Central

### PR DESCRIPTION
**Pull request by jmfinelli**
Addresses JBTM-4002

### CODE REVIEW

**Review by tomjenkinson — APPROVED** [review]
LGTM - thank you @jmfinelli !

**Review by xstefank — APPROVED** [review]

File: proxy/pom.xml [hunk 1]
```diff
@@ -19,32 +19,19 @@
 
   <modules>
     <module>api</module>
-    <module>test</module>
```
Comment (xstefank) [inline proxy/pom.xml:22]: why?

**Review by xstefank — COMMENTED** [review]
I don't like the way how the test module is excluded. Sorry if I missed it before.

File: proxy/pom.xml (hunk 1 above)
Comment (jmfinelli) [reply proxy/pom.xml:22]: You didn't miss it before, I introduce it now. Sorry for the confusion.

Comment (jmfinelli) [reply proxy/pom.xml:22]: > sorry, do it like this:
>
> https://github.com/smallrye/smallrye-health/blob/main/pom.xml#L260-L271
>
> basically if not performing release, not if not skipping tests.

I would like to avoid introducing another property. I think that `-DskipTests` is a good fit in our release process.

Comment (xstefank) [reply proxy/pom.xml:22]: @jmfinelli no, `-DskipTests` means compile tests but don't run them. So you are not correct with your assumption.

you should include the test module if there is not -Prelease or -PnexusStagingDeployment, this is why it would be better to have property flags like -Drelease and -DreleaseStaging that can be negated to include test if none of them is specified.

**Review by xstefank — APPROVED** [review]
